library/src/main/java/net/grandcentrix/tray/provider/TrayContract.java

```diff
@@ -120,8 +121,20 @@ private static Uri generateContentUri(@NonNull final Context context, final Stri
 
     @NonNull
     private static String getAuthority(@NonNull final Context context) {
-        return TextUtils.isEmpty(sTestAuthority) ?
-                context.getString(R.string.tray__authority) :
-                sTestAuthority;
+        if (!TextUtils.isEmpty(sTestAuthority)) {
+            return sTestAuthority;
+        }
+        for (PackageInfo pack : context.getPackageManager().getInstalledPackages(PackageManager.GET_PROVIDERS)) {
+            ProviderInfo[] providers = pack.providers;
+            if (providers != null) {
+                for (ProviderInfo provider : providers) {
+                    if (TextUtils.equals(provider.authority,
+                            context.getApplicationContext().getPackageName() + ".tray")) {
+                        return provider.authority;
```

Comment:
Checkout [this test](https://github.com/b95505017/tray/blob/eeb5ef218c0a15c6a98beb9e72dd6045e4d24c6c/library/src/androidTest/java/net/grandcentrix/tray/provider/TrayContractTest.java).
You can get the `Uri` via `generateContentUri` which should contain the `authority`.